executor/builder.go

```diff
@@ -1276,16 +1289,17 @@ type dataReaderBuilder struct {
 	*executorBuilder
 }
 
-func (builder *dataReaderBuilder) buildExecutorForDatums(goCtx goctx.Context, datums [][]types.Datum) (Executor, error) {
+func (builder *dataReaderBuilder) buildExecutorForIndexJoin(goCtx goctx.Context, datums [][]types.Datum,
+	IndexRanges []*ranger.IndexRange, offsetsMap []int) (Executor, error) {
 	switch v := builder.Plan.(type) {
-	case *plan.PhysicalIndexReader:
-		return builder.buildIndexReaderForDatums(goCtx, v, datums)
 	case *plan.PhysicalTableReader:
 		return builder.buildTableReaderForDatums(goCtx, v, datums)
+	case *plan.PhysicalIndexReader:
+		return builder.buildIndexReaderForIndexJoin(goCtx, v, datums, IndexRanges, offsetsMap)
```

Comment:
It may not be for IndexJoin, can be for index lookup reader.

Reply:
Double read haven't called this method.